lib/tether/toolbox.go

```diff
@@ -56,10 +61,63 @@ func (t *Toolbox) Stop() error {
 }
 
 // Reload implementation of the tether.Extension interface
-func (*Toolbox) Reload(config *ExecutorConfig) error {
+func (t *Toolbox) Reload(config *ExecutorConfig) error {
+	t.config = config
+	return nil
+}
+
+// InContainer configures the toolbox to run within a container VM
+func (t *Toolbox) InContainer() *Toolbox {
+	t.PowerCommand.Halt.Handler = t.halt
+
+	return t
+}
+
+func (t *Toolbox) kill(name string) error {
+	session := t.config.Sessions[t.config.ID]
+
+	sig := new(msgs.SignalMsg)
+	err := sig.FromString(name)
+	if err != nil {
+		return err
+	}
+
+	num := syscall.Signal(sig.Signum())
+
+	log.Infof("sending signal %s (%d) to %s", sig.Signal, num, session.ID)
+
+	if err := session.Cmd.Process.Signal(num); err != nil {
+		return fmt.Errorf("failed to signal %s: %s", session.ID, err)
+	}
+
 	return nil
 }
 
+func (t *Toolbox) halt() error {
+	session := t.config.Sessions[t.config.ID]
```

Comment:
AFAICT it would only be empty if someone manually removed it from the .vmx, but can add a check anyways.